batch/batch/cloud/gcp/driver/zones.py

```diff
@@ -89,10 +93,16 @@ def choose_location(
         data_disk_size_gb: int,
         preemptible: bool,
         regions: List[str],
+        machine_type,
     ) -> str:
         zone_weights = self.compute_zone_weights(cores, local_ssd_data_disk, data_disk_size_gb, preemptible, regions)
 
         zones = [zw.zone for zw in zone_weights]
+        machine_family = machine_type.split("-")[0]
+        if machine_family in self._machine_family_valid_zones:
```

Comment:
Actually, we should change this behavior to always assume that the machine_family is always a key in `self._machine_family_valid_zones`. If it isn't, then we should set zones and zone_weights to be equal to `[]`.